Lib/dataclasses.py

```diff
@@ -222,6 +222,29 @@ def __repr__(self):
 # https://bugs.python.org/issue33453 for details.
 _MODULE_IDENTIFIER_RE = re.compile(r'^(?:\s*(\w+)\s*\.)?\s*(\w+)')
 
+# Types for which deepcopy(obj) is known to return obj unmodified
+# Used to skip deepcopy in asdict and astuple for performance
```

Comment:
I think this comment is overly-focused on deepcopy.

It is important for semantic consistency that all of these types be ones for whom deepcopy is identity, but it is not important that we include here the full list of types for which that is true, and "skipping deepcopy" does not fully describe the optimization that we apply to these types.
```suggestion
# Atomic immutable types which don't require any recursive handling and for which deepcopy
# returns the same object. We can provide a fast-path for these types in asdict and astuple.
```